### PR TITLE
[CALCITE] Fixed validation for FIRST_VALUE/LAST_VALUE

### PR DESCRIPTION
**Pull request by AndrewPochapsky**
- Added simple validation for SqlNullTreatmentModifier
- Changed number of params in FIRST_VALUE function to two (second optional)

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java [hunk 1]
```diff
@@ -6015,6 +6020,10 @@ public RelDataType visit(SqlIntervalQualifier intervalQualifier) {
       return typeFactory.createSqlIntervalType(intervalQualifier);
     }
 
+    @Override public RelDataType visit(SqlNullTreatmentModifier nullTreatmentModifier) {
```
Comment (dasch-1) [inline core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java:6023]: nit: Line wrap

File: core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java [hunk 2]
```diff
@@ -6045,6 +6054,10 @@ public RelDataType visit(SqlIntervalQualifier intervalQualifier) {
       return expandedExpr;
     }
 
+    @Override public SqlNode visit(SqlNullTreatmentModifier nullTreatmentModifier) {
```
Comment (dasch-1) [inline core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java:6057]: nit: Line wrap

File: core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java [hunk 3]
```diff
@@ -6281,6 +6298,10 @@ protected SqlNode visitScoped(SqlCall call) {
       return super.visit(id);
     }
 
+    @Override public SqlNode visit(SqlNullTreatmentModifier nullTreatmentModifier) {
```
Comment (dasch-1) [inline core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java:6301]: nit: Line wrap

File: core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java [hunk 4]
```diff
@@ -6648,6 +6669,10 @@ public SqlNode go(SqlNode node) {
       return ImmutableSet.of();
     }
 
+    @Override public Set<String> visit(SqlNullTreatmentModifier nullTreatmentModifier) {
```
Comment (dasch-1) [inline core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java:6672]: nit: Line wrap

File: core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java [hunk 5]
```diff
@@ -2331,6 +2332,10 @@ protected void convertMatchRecognize(Blackboard bb, SqlCall call) {
             return rexBuilder.makeLiteral(id.getSimple());
           }
 
+          @Override public RexNode visit(SqlNullTreatmentModifier nullTreatmentModifier) {
```
Comment (dasch-1) [inline core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java:2335]: nit: Line wrap

File: core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java [hunk 6]
```diff
@@ -5036,6 +5041,10 @@ public RexNode visit(SqlIntervalQualifier intervalQualifier) {
       return convertInterval(intervalQualifier);
     }
 
+    @Override public RexNode visit(SqlNullTreatmentModifier nullTreatmentModifier) {
```
Comment (dasch-1) [inline core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java:5044]: nit: Line wrap

File: parsing/dialects/intermediate/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ValidatorTest.java [hunk 7]
```diff
@@ -42,6 +42,38 @@
     sql(sql).rewritesTo(expected);
   }
 
+  @Test public void testFirstValueRewrite() {
+    String sql = "SELECT FIRST_VALUE (foo) OVER (PARTITION BY (foo)) FROM bar";
+    String expected = "SELECT FIRST_VALUE(`BAR`.`FOO`) OVER (PARTITION BY `BAR`.`FOO`)\n"
```
Comment (dasch-1) [inline parsing/dialects/intermediate/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ValidatorTest.java:47]: nit: Line wrap (here and elsewhere in the file)

File: core/src/main/java/org/apache/calcite/sql/fun/SqlFirstLastValueAggFunction.java [hunk 8]
```diff
@@ -48,7 +48,7 @@ public SqlFirstLastValueAggFunction(SqlKind kind) {
         kind,
         ReturnTypes.ARG0_NULLABLE_IF_EMPTY,
         null,
-        OperandTypes.ANY,
+        OperandTypes.ANY_OPTIONAL_ANY,
```
Comment (dstekol) [inline core/src/main/java/org/apache/calcite/sql/fun/SqlFirstLastValueAggFunction.java:51]: Curious, what does this do?

Comment (dasch-1) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlFirstLastValueAggFunction.java:51]: I think it specifies that the first value can be any data type and the second value can also be of any data type, but may also be optionally omitted.

**Pull request merged**